src/main/java/org/cryptomator/common/SubstitutingProperties.java

```diff
@@ -0,0 +1,69 @@
+package org.cryptomator.common;
+
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Pattern;
+
+public class SubstitutingProperties extends PropertiesDecorator {
+
+	private static final Pattern TEMPLATE = Pattern.compile("@\\{(\\w+)}");
+
+	private final Map<String, String> env;
+
+	public SubstitutingProperties(Properties props, Map<String, String> systemEnvironment) {
+		super(props);
+		this.env = systemEnvironment;
+	}
+
+	@Override
+	public String getProperty(String key) {
+		var value = delegate.getProperty(key);
+		if (key.startsWith("cryptomator.") && value != null) {
+			return process(value);
+		} else {
+			return value;
+		}
+	}
+
+	@Override
+	public String getProperty(String key, String defaultValue) {
+		var result = getProperty(key);
+		return result != null ? result : defaultValue;
+	}
+
+	//visible for testing
```

Comment:
we can use `@org.jetbrains.annotations.VisibleForTesting` as well

Reply:
Added in https://github.com/cryptomator/cryptomator/pull/2957/commits/29d63a0a83b7399ba65bd07ff57bf28e5a770ffd